client/src/components/UserDetailsForm/UserDetailsForm.js

```diff
@@ -0,0 +1,36 @@
+/* eslint-disable react/prop-types */
+import React from 'react';
+import { Grid, Typography } from '@mui/material';
+import { TextField } from 'formik-mui';
+import { Field } from 'formik';
+
+const UserDetailsForm = ({ checkoutFormModel }) => {
+  const { firstName, lastName, email, phone } = checkoutFormModel;
+
+  return (
+    <>
+      <Typography variant="h3" align="center" sx={{ mb: '20px' }}>
+        User Details
+      </Typography>
+
+      <Grid container spacing={2}>
+        <Grid item xs={12} laptop={6}>
+          <Field component={TextField} name={firstName.name} label={firstName.label} fullWidth />
+        </Grid>
+
+        <Grid item xs={12} laptop={6}>
+          <Field component={TextField} name={lastName.name} label={lastName.label} fullWidth />
+        </Grid>
+
+        <Grid item xs={12} laptop={6}>
+          <Field component={TextField} name={email.name} label={email.label} fullWidth />
+        </Grid>
+
+        <Grid item xs={12} laptop={6}>
+          <Field component={TextField} name={phone.name} label={phone.label} fullWidth />
+        </Grid>
+      </Grid>
+    </>
```

Comment:
```suggestion
const UserDetailsForm = ({ checkoutFormModel }) => {
  const { firstName, lastName, email, phone } = checkoutFormModel;
  const data = [firstName, lastName, email, phone];
  return (
    <>
      <Typography variant="h3" align="center" sx={{ mb: '20px' }}>
        User Details
      </Typography>

      <Grid container spacing={2}>
          {data.map((el) => (
          <Grid item xs={12} laptop={6}>
            <Field component={TextField} name={el.name} label={el.label} fullWidth />
          </Grid>
        ))}
    </>
```

Reply:
If you agree to use the above method, please apply it in ShippingAddressForm and PaymentForm

Reply:
dont agree, since these are gird items and sizing is controlled individually by props for different breakpoints, for better readability and control over each item, better to be represented individually not through a map method.